library/Zend/Code/Scanner/ClassScanner.php

```diff
@@ -480,6 +505,142 @@ public function hasProperty($name)
         return false;
     }
 
+
+    public function getTraits()
+    {
+        if(empty($this->traits)) {
+            //get list of traitNames
+            $traitNames = $this->getTraitNames();
+            foreach($traitNames as $traitName) {
+                /*
+                 * @todo test if this works when trait is included or required instead of autoloaded?
+                 */
+                $fileName = str_replace("\\", "/", $traitName) .".php";
```

Comment:
You could do something like this:

```php
$r = new ReflectionClass($traitName);
if (! $r->isTrait()) {
    throw new Exception\RuntimeException('Non-trait detected as a trait: ' . $traitName);
}
$fileName = $r->getFileName();
```

I'll give that a try shortly to see if it works.